nucypher/datastore/datastore.py

```diff
@@ -14,269 +14,222 @@
-        deleted_records = 0
-        if result.count() > 0:
-            deleted_records = result.delete()
-        self.__commit(session=session)
-        return deleted_records
-
-    #
-    # Work Orders
-    #
-
-    def save_workorder(self,
-                       bob_verifying_key: UmbralPublicKey,
-                       bob_signature: Signature,
-                       arrangement_id: bytes,
-                       session=None
-                       ) -> Workorder:
+        Specifically, this allows us to match partial keys to specific keys,
+        where the `Datastore` will _always_ return specific keys, but queries
+        will almost always be partial keys.
         """
-        Adds a Workorder to the keystore.
-        """
-        session = session or self._session_on_init_thread
-
-        # Get or Create Bob Verifying Key
-        fingerprint = fingerprint_from_key(bob_verifying_key)
-        key = session.query(Key).filter_by(fingerprint=fingerprint).first()
-        if not key:
-            key = self.add_key(key=bob_verifying_key)
-
-        new_workorder = Workorder(bob_verifying_key_id=key.id,
-                                  bob_signature=bob_signature,
-                                  arrangement_id=arrangement_id)
-
-        session.add(new_workorder)
-        self.__commit(session=session)
-        return new_workorder
-
-    def get_workorders(self,
-                       arrangement_id: bytes = None,
-                       bob_verifying_key: bytes = None,
-                       session=None
-                       ) -> List[Workorder]:
-        """
-        Returns a list of Workorders by HRAC.
-        """
-        session = session or self._session_on_init_thread
-        query = session.query(Workorder)
-
-        if not arrangement_id and not bob_verifying_key:
-            workorders = query.all()  # Return all records
-
-        else:
-
-            # Return arrangement records
-            if arrangement_id:
-                workorders = query.filter_by(arrangement_id=arrangement_id)
-
-            # Return records for Bob
-            else:
-                fingerprint = fingerprint_from_key(bob_verifying_key)
-                key = session.query(Key).filter_by(fingerprint=fingerprint).first()
-                workorders = query.filter_by(bob_verifying_key_id=key.id)
+        other_key = DatastoreKey.from_bytestring(key_bytestring)
+        return self.record_type == (other_key.record_type or self.record_type) and \
+               self.record_field == (other_key.record_field or self.record_field) and \
+               self.record_id == (other_key.record_id or self.record_id)
 
-            if not workorders:
-                raise NotFound
 
-        return list(workorders)
-
-    def del_workorders(self, arrangement_id: bytes, session=None) -> int:
-        """
-        Deletes a Workorder from the Keystore.
-        """
-        session = session or self._session_on_init_thread
+class Datastore:
```

Comment:
This full path has become `nucypher.datastore.datastore.Datastore` - which is a bit repetitive.